packages/python/genai_prices/types.py

```diff
@@ -53,6 +74,29 @@
 ]
 
 
+def find_provider_by_id(providers: list['Provider'], provider_id: str) -> 'Provider | None':
```

Comment:
you shouldn't need the quotes in type hints, also please move this into `sources.py`, it doesn't make sense here.